packages/directory/addon/components/dir-table.js

```diff
@@ -100,39 +101,55 @@ export default Component.extend({
    * @property {Object} table - Used by ember-light-table to create the table
    */
   table: computed('model', function() {
-    let table = new Table(this.get('columns'), this.get('model'), {
+    let table = new Table(get(this, 'columns'), get(this, 'model'), {
       rowOptions: {
         classNames: 'dir-table__row'
       }
     });
 
-    let sortColumn = table.get('allColumns').findBy('sortByKey', this.get('sortBy'));
+    let sortColumn = table.get('allColumns').findBy('sortByKey', get(this, 'sortBy'));
 
     if (sortColumn) {
       sortColumn.setProperties({
         sorted: true,
-        ascending: this.get('sortDir') !== 'desc'
+        ascending: get(this, 'sortDir') !== 'desc'
       });
     }
 
     return table;
   }),
 
+  _getNextSort(column) {
+    let sortBy = get(this, 'sortBy'),
+      nextSortBy = get(column, 'sortByKey'),
+      nextSortDir;
+
+    if (sortBy === nextSortBy) {
+      nextSortDir = get(column, 'ascending') ? 'asc' : 'desc';
+    } else {
+      nextSortDir = get(column, 'sortDescFirst') ? 'desc' : 'asc';
+    }
+
+    return {
+      sortBy: nextSortBy,
+      sortDir: nextSortDir
+    };
+  },
+
   actions: {
     /**
      * @action onColumnClick
      * @param {Object} column
      */
     onColumnClick(column) {
       if (column.sorted) {
-        let onColumnClick = this.get('onColumnClick');
-
-        if (typeof onColumnClick !== 'function') return;
+        let onColumnClick = get(this, 'onColumnClick');
 
-        let sortBy = get(column, 'sortByKey'),
-          sortDir = get(column, 'ascending') ? 'asc' : 'desc';
+        if (typeof onColumnClick !== 'function') {
```

Comment:
Could we just move the `onColumnClick(this._getNextSort(column));` inside the conditional?

`if (typeof onColumnClick === 'function') {`
`    onColumnClick(this._getNextSort(column));`
`}`

Reply:
done, it did look weird